vision_datasets/common/data_manifest.py

```diff
@@ -288,6 +289,41 @@ def sample_few_shot_subset(self, num_samples_per_class, random_seed=0):
 
         return DatasetManifest(sampled_images, self.labelmap, self.data_type)
 
+    def sample_subset_by_ratio(self, sampling_ratio):
+        """
+        Sample a dataset so that each labels appears by at least the given sampling_ratio. In case of multiclass dataset, the number of sampled images will be N * sampling_ratio.
+        For multilabel or object detection datasets, the total number of images will be bigger than that.
+
+        Args:
+            sampling_ratio (float): sampling raito. must be 0 < x < 1.
+
+        Returns:
+            A sampled dataset (DatasetManifest)
+        """
+        assert 0 < sampling_ratio < 1
+
+        if self.is_multitask:
+            labels = [[self._get_cid(c, t) for t, t_labels in image.labels.items() for c in t_labels] for image in self.images]
+        else:
+            labels = [[self._get_cid(c) for c in image.labels] for image in self.images]
+
+        # Create a dict {label_id: [image_id, ...], ...}
+        # Note that image_id can be included multiple times if the dataset is multilabel, objectdetection, or multitask.
+        label_image_map = collections.defaultdict(list)
+        for i, image_labels in enumerate(labels):
+            if not image_labels:
```

Comment:
there is a _is_negative() function which takes care of multitask case as well

Reply:
this "labels" variable is a 2-d list constructed above. image_labels will be a 1-d empty list even on multitask case.

Reply:
got it